packages/data-connect/src/api/DataConnect.ts

```diff
@@ -116,6 +122,10 @@ export class DataConnect {
       this._isUsingGeneratedSdk = true;
     }
   }
+  // @internal
+  _setCallerSdkType(callerSdkType: CallerSdkType): void {
+    this._callerSdkType = callerSdkType;
```

Comment:
This should call a setter on the transport class. 
The problem that I'm seeing is that if `_setCallerSdkType` gets called after the ref has been created, the `_callerSdkType` doesn't get passed to the transportClass, as that happens whenever the ref has been created